### PR TITLE
Hull, Domain and Geometry Shaders for WindowsDX

### PR DESCRIPTION
**Pull request by cpt-max**
This pull request is based on [JSandusky's MonoGame branch](https://github.com/JSandusky/MonoGame).
It adds the missing DX11 shader stages to the framework and to the content pipeline (mgfxc). 
I created this [MonoGame sample project](https://github.com/cpt-max/ShaderTest) to test the new stages.

The MGFXVersion has been raised to 10 which means shader content need to be rebuilt. 
The changes shouldn't break other platforms (theoretically), but they add some non-functional elements to the framework, like for example new PrimitiveTypes that only work with hull shaders. I could #if them out of those platforms for now, but I'd prefer to make them functional instead :)

The DX unit tests didn't work for me, every test that needs a DX device fails to create one. What am I missing? OpenGL unit tests work fine.

### CODE REVIEW

**Comment by tomspilman** [comment]
So for sure this needs to sit at least until 3.8 is out the door... we'll get this into 3.9.

File: MonoGame.Framework/Graphics/Shader/ShaderStage.cs [hunk 1]
```diff
@@ -4,5 +4,8 @@ internal enum ShaderStage
     {
         Vertex,
         Pixel,
+        Hull,
+        Domain,
+        Geometry,
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Shader/ShaderStage.cs:9]: We should document exactly what these are even with it being internal.  Will help when we port to new platforms and also when we move to make the Shader API public.

**Comment by tomspilman** [comment]
@cpt-max - You should add a new effect processor test for these new shader stages.  These were moved here:

https://github.com/MonoGame/MonoGame/blob/develop/Tools/MonoGame.Tools.Tests/EffectProcessorTests.cs

You will need to put a `#if DIRECTX` around it keep it from running the tests on the Mac.

**Comment by georg-eckert-zeiss** [comment]
@cpt-max - I had a look into your PR. Looks very promising, great work! 

I noticed that `PrimitiveType` does not contain `LineListAdj`, `LineStripAdj`, `TriangleListAdj` and `TriangleStripAdj` - on purpose?

Cheers,
Georg

**Comment by cpt-max** [comment]
> I noticed that `PrimitiveType` does not contain `LineListAdj`, `LineStripAdj`, `TriangleListAdj` and `TriangleStripAdj`

I'll add those when I update this PR.

**Comment by cpt-max** [comment]
@tomspilman I added the documentation and unit tests you requested. I don't think the DIRECTX symbol is defined though, so those tests don't run.
@georg-eckert-zeiss the adjacency primitive types are now available.

**Comment by tomspilman** [comment]
Thanks @cpt-max .  I'll find some time to review this in more detail this week.

**Comment by georg-eckert-zeiss** [comment]
> @georg-eckert-zeiss the adjacency primitive types are now available.

Great : ) Thank you. Really looking forward to this.

**Comment by cpt-max** [comment]
I didn't realize at first that some files in MGFXC are supposed to be autogenerated by TinyPG, rather than changed manually. With the latest commit they are autogenerated again.

**Comment by cpt-max** [comment]
Closing this PR, as it's contained in https://github.com/MonoGame/MonoGame/pull/7533

**Pull request closed without merging**